internal/metadata/store.go

```diff
@@ -0,0 +1,53 @@
+package metadata
+
+type BlockHeightUpdate struct {
+	GroupID     string
+	BlockHeight uint64
+}
+
+type ChainMetadataStore struct {
+	blockHeightChannel <-chan BlockHeightUpdate
+	maxHeightByGroupID map[string]uint64
+	globalMaxHeight    uint64
+}
+
+func NewChainMetadataStore(blockHeightChannel <-chan BlockHeightUpdate) *ChainMetadataStore {
+	return &ChainMetadataStore{
+		blockHeightChannel: blockHeightChannel,
+		maxHeightByGroupID: make(map[string]uint64),
+	}
+}
+
+func (c *ChainMetadataStore) Start() {
+	go func() {
+		for {
+			var update = <-c.blockHeightChannel
+			c.globalMaxHeight = max(c.globalMaxHeight, update.BlockHeight)
+			c.updateHeightForGroup(update.GroupID, update.BlockHeight)
+		}
+	}()
+}
+
+func (c *ChainMetadataStore) updateHeightForGroup(groupID string, currentBlockHeight uint64) {
+	var oldHeight = c.maxHeightByGroupID[groupID]
+
+	var newMaxHeight = max(oldHeight, currentBlockHeight)
```

Comment:
What are your thoughts on using `var` vs using `:=` syntax? AFAIK this repo only has `:=` syntax where possible

Reply:
Ah yep, good question. I hit an issue with accidental variable shadowing very early on as I was experimenting with Go - something like this:

```
q := make([]int)
for len(q) > 0 {
  firstItem, q := q[0], q[1:]
}
```

which went into an infinite loop and took me a decent amount of time to notice that the `:=` totally changes the behaviour of the code (creating a new queue inside the loop body while keeping the outer one intact).

So when writing new code, I was initially planning to only use the `var x = ...` syntax. However, as I was started modifying existing functions, it seemed nicer to stay consistent with the surrounding code. That's how I ended up with `var x = ` in some places and `x := ...` in others.

I'll post a thing in Slack to see if we can form a consensus around which one to use.

Reply:
I converted all the var declarations to short form as I think that's the decision we converged on.